salt/tests/unit/modules/test_cordon.py

```diff
@@ -0,0 +1,175 @@
+        self.assertEqual(metalk8s_cordon.__virtual__(), 'metalk8s_kubernetes')
+
+    def test_check_deps_success(self):
+        """
+        Tests the availability of `metalk8s_kubernetes.update_object` module
+        in salt
+        """
+        with patch.dict(metalk8s_cordon.__salt__, {
+                'metalk8s_kubernetes.update_object': MagicMock()
+        }):
+            metalk8s_cordon._check_deps()
+
+    def test_check_deps_fail(self):
+        """
+        Tests the unavailability of `metalk8s_kubernetes.update_object` module
+        in salt
+        """
+        self.assertRaisesRegexp(
+            CommandExecutionError,
+            "'metalk8s_kubernetes.update_object' is not available",
+            metalk8s_cordon._check_deps)
+
+    @parameterized.expand([
+        (None,),
+        (True,),
+    ])
+    def test_cordon_node(self, unschedulable, **kwargs):
+        """
+        Tests the return of `cordon_node` function
+        """
+        def _update_object_mock(patch, **_):
+            return salt.utils.dictupdate.update(node_obj, patch)
+
+        node_obj = self.node_object(unschedulable=unschedulable)
+        node_update_mock = MagicMock(side_effect=_update_object_mock)
+        salt_dict = {'metalk8s_kubernetes.update_object': node_update_mock}
+        with patch.dict(metalk8s_cordon.__salt__, salt_dict):
+            self.assertTrue(
+                metalk8s_cordon.cordon_node(
+                    node_name="bootstrap",
+                    **kwargs
+                )['spec']['unschedulable']
+            )
+            self.assertEqual(
+                node_obj,
+                metalk8s_cordon.cordon_node(
+                    node_name="bootstrap",
+                    **kwargs
+                )
+            )
+            node_update_mock.assert_called()
+
+    @parameterized.expand([
+        (None, False),
+        (True, False),
+        (False, False),
+    ])
+    def test_uncordon_with_no_unschedulable_taints(
+        self, unschedulable, with_taints
+    ):
+        """
+        Tests the return of `uncordon_node` function when no unschedulable
+        taints is set
+        """
+        def _update_object_mock(patch, **_):
+            return salt.utils.dictupdate.update(node_obj, patch)
+
+        node_obj = self.node_object(
+            unschedulable=unschedulable, with_taints=with_taints
+        )
+        get_object_mock = MagicMock(return_value=node_obj)
+        node_update_mock = MagicMock(side_effect=_update_object_mock)
+
+        salt_dict = {
+            'metalk8s_kubernetes.get_object': get_object_mock,
+            'metalk8s_kubernetes.update_object': node_update_mock
+        }
+        with patch.dict(metalk8s_cordon.__salt__, salt_dict):
+            self.assertFalse(
+                metalk8s_cordon.uncordon_node(
+                    node_name="bootstrap",
+                )['spec']['unschedulable']
+            )
+            self.assertEqual(
+                node_obj,
```

Comment:
No it's wrong, here you check nothing since node_obj is the dict you update, you want one "uncordoned" node

```suggestion
                self.node_object(unschedulable=False),
```